pkg/syncer/pvcsi_fullsync.go

```diff
@@ -351,29 +363,62 @@ func AddNodeAffinityRulesOnPV(ctx context.Context, metadataSyncer *metadataSyncI
 			oldData, err := json.Marshal(pv)
 			if err != nil {
 				log.Errorf("failed to marshal pv: %v, Error: %v", pv, err)
-				continue
+				return err
 			}
 			newPV := pv.DeepCopy()
 			newPV.Spec.NodeAffinity = GenerateVolumeNodeAffinity(csiAccessibleTopology)
 			newData, err := json.Marshal(newPV)
 			if err != nil {
 				log.Errorf("failed to marshal updated PV with node affinity rules: %v, Error: %v", newPV, err)
-				continue
+				return err
 			}
 
 			patchBytes, err := strategicpatch.CreateTwoWayMergePatch(oldData, newData, pv)
 			if err != nil {
 				log.Errorf("error Creating two way merge patch for PV %q with error : %v", pv.Name, err)
-				continue
+				return err
 			}
 			_, err = k8sClient.CoreV1().PersistentVolumes().Patch(
 				context.TODO(), pv.Name, types.StrategicMergePatchType, patchBytes, metav1.PatchOptions{})
 			if err != nil {
 				log.Errorf("error patching PV %q error : %v", pv.Name, err)
-				continue
+				return err
 			}
 			log.Infof("patched PV: %v with node affinity %v", pv.Name, newPV.Spec.NodeAffinity)
 		}
+		return nil
+	}
+
+	for _, pv := range pvList {
+		_ = addNodeAffinityOnPVInternal(pv)
+	}
+
+	// Check if there are any PVs on which we didn't add node affinity rules yet, as topology annotation
+	// was missing from associated supervisor PVC. We will iterate over all such PVs again and will check if
+	// PVC annotation is added now. We will retry this until all PVs get node affinity rules added or until
+	// timeout of 5 minutes is reached.
+	timeoutCtx, cancel := context.WithTimeout(context.Background(), timeoutAddNodeAffinityOnPVs)
+	defer cancel()
+	for len(pvsWithoutSupervisorPvcTopologyAnnotation) != 0 {
```

Comment:
if possible, can we add a separate function for this for loop code block? also separate function for addNodeAffinityOnPVInternal outside of AddNodeAffinityRulesOnPV?

Reply:
AddNodeAffinityRulesOnPV is a small function, so keeping inline function and newly added for loop in same function to keep all flow at same place.